### ISSUE TITLE
Outline Adversary Models

### ISSUE THREAD

**Issue by sventuerpe** [corona-warn-app/cwa-documentation#123]
## What is missing

There should be a rough classification and high-level description of adversary types in relation to the architecture, the stakeholders and the environment of CWA. The description of each adversary type should remain concise and focus on capabilities and constraints with respect to which parts of the CWA system can observe or manipulate. I imagine something along these lines:

1. A rogue app user controlling one or more instances of the app through App and mobile OS user interfaces.
1. A rogue participant running their own software on one or more unconstrained devices, interacting with other devices and servers in any way supported by the respective protocol implementations.
1. An adversary controlling the server side.
1. An adversary controlling the mobile OS and/or app store.
1. An adversary capable of recording Internet traffic (a) at a small scale, (b) at a large scale.
1. …

Where the capabilities (constraints) of one adversary are a superset (subset) of those of another, this should be mentioned. For example, control of the mobile OS or app store (4) implies the ability to become – or make others – a rogue participant (2). 

## Why should it be included

A list of adversary models:

- _Facilitates constructive discussions of abuse and attack scenarios_ – A list of pertinent adversary models helps clarify the assumptions underlying a threat scenario.
- _Supports risk assessment_ – One can estimate for each scenario what it would take for an adversary to attain the respective capabilities or to overcome the constraints.
- _Complements abuser stories (#71),_ which should focus on adversaries’ objectives, with a capability/constraint perspective.

## Where should it be included

Security documentation or high-level design document.

**Comment by sventuerpe** [comment]
Just a pointer: #147 has a discussion on collusion attacks, which seems related

**Comment by haxxbard** [comment]
Thanks for bringing this up, @sventuerpe.
We already published the [security overview](https://github.com/corona-warn-app/cwa-documentation/blob/master/overview-security.md) which contains a high level risk assessment a while ago. 
The topics you brought up were discussed during our threat modeling workshops and are therefore already incorporated into the high level risk assessment we published. We do not plan to publish further documentation wrt to your request, yet; hence I'll close this issue.


Thanks
haxxbard
